extension/observer/cfgardenobserver/README.md

```diff
@@ -20,34 +20,54 @@ The `cfgarden_observer` looks at the current host to discover Garden containers.
 ```yaml
 extensions:
   cfgarden_observer:
-    # url of the Garden socket, defaults to unix:///var/vcap/data/garden/garden.sock 
-    endpoint: my/path/to/garden.sock
-    # determines how often to look for changes in endpoints.
     refresh_interval: 30s
+    cache_sync_interval: 10m
+    include_app_labels: true
+    garden:
+      endpoint: my/path/to/garden.sock
+    cloud_foundry:
+      endpoint: https://api.cf.mydomain.com
+      auth_type: client_credentials
+      client_id: myclientid
+      client_secret: myclientsecret
 
 receivers:
   receiver_creator:
     watch_observers: [cfgarden_observer]
     receivers:
       prometheus_simple:
-        rule: type == "container" && name == "myapp"
+        rule: type == "container" && labels["prometheus.io/scrape"] == "true" 
         config:
           metrics_path: /metrics
           endpoint: '`endpoint`'
 ```
 
 ### Configuration
-
-| Name             | Type   | Default                                  | Docs                                                   |
-|------------------|--------|------------------------------------------|--------------------------------------------------------|
-| refresh_interval | string | 60s                                      | Determines how often to look for changes in endpoints. |
-| endpoint         | string | unix:///var/vcap/data/garden/garden.sock | The endpoint to connect to the Garden API.             |
+| Name                        | Type   | Default                                                                | Description                                                        |
+| --------------------------- | ------ | ---------------------------------------------------------------------- | ------------------------------------------------------------------ |
+| refresh_interval            | string | 1m                                                                     | Determines how often to look for changes in endpoints.             |
+| cache_sync_interval         | string | 5m                                                                     | Determines how often app metadata cache is refreshed               |
+| include_app_labels          | bool   | false                                                                  | Determines whether or not app labels get added to container labels |
+| garden.endpoint             | string | /var/vcap/data/garden/garden.sock                                      | Path to garden socket.                                             |
+| cloud_foundry.endpoint      | string | none. required when `include_app_labels` is set to `true` | CloudFoundry API endpoint                                          |
+| cloud_foundry.auth_type     | string | none. required when `include_app_labels` is set to `true` | Authentication type, one of: user_pass, client_credentials, token  |
```

Comment:
What do you think of moving authentication options under a specific config section for each? Something like:
```
auth_type: user_pass
user_pass:
  username: "a"
  password: "..."
```
I'm wondering if it would help clarify that each of these options are only relevant for a specific authentication type. If a config then contains options irrelevant to the `auth_type` chosen, they'd just be ignored. Do you think that would be a more clear way of handling this?

Reply:
hmm it's a good point. I tried implementing your suggestion and after seeing it I personally feel like it's more clunky to have them in separate sections, I would rather make it clearer in the readme.

The following, for example, looks a bit odd to me:
```
auth_type: token
token:
  access_token: "a"
  refresh_token: "..."
```

Other namings I tried for the token stuff also seem odd to me, but maybe this is just a personal preference. Ultimately I value your opinion more so if you still think it's best of course I will do it that way!

Reply:
That's fair, I can see that being a bit clunky. Would it make sense to put all authentication options under the same header?

```
authentication:
  type: token
  access_token: "a"
  refresh_token: "..."
```
It could be `auth` instead of `authentication`, but do you think that would help? I just want to avoid having such a flat layout when all of these options are related/dependent on each other somehow.

Reply:
good idea, I've changed it now